Created base folder structure and classes depensences

IhorBerehovskyi opened the pull request.

IhorBerehovskyi committed:
Created base folder structure and classes depensences

Sage-Cat commented on server/src/Application/Application.hpp:
```diff
@@ -0,0 +1,31 @@
+
+#include <sqlite3.h>
+#include <functional> 
+#include "../Data/Data.hpp"
```
Don't use relative path

Sage-Cat commented on server/src/Data/Data.hpp:
```diff
@@ -0,0 +1,11 @@
+    //some data from application
+};
+
+using Callback = std::function<void(std::string, Data)>;
```
Not Callback. Give some appropriate name, like `ProcessDataCallback` or `PrepareResponseCallback`

IhorBerehovskyi replied:
Done

Sage-Cat commented on server/src/Application/Application.cpp:
```diff
@@ -0,0 +1,19 @@
+
+Application::Application(CO2Sensor &co2Sensor, SQLiteDatabase& sqliteDatabase) : 
+                                                                    sensor(co2Sensor),
+                                                                    database(sqliteDatabase){
```
Use clang-format for formatting code.
There is VS Code extension for it.

Sage-Cat commented on server/src/Data/Data.hpp:
```diff
@@ -0,0 +1,11 @@
+
+struct Data
+{
+    //some data from application
```
let it be just `std::string currentCO2Level`

Sage-Cat replied:
I wanted you to create variable here

IhorBerehovskyi replied:
Done

Sage-Cat commented on server/src/Session.hpp:
```diff
@@ -0,0 +1,27 @@
+    ip::tcp::socket socket_;
+    char data_[512];
+
+    AppCallback appCallback;
```
Same for this. Make better name

Sage-Cat committed:
Add CMake configs and update README

We need to have some build system

Signed-off-by: Sage-Cat <sagecatbox@gmail.com>

Sage-Cat reviewed (changes requested):
Fix all the build issues:
```
╭─sagecat@sage-pc ~/Projects/measure_CO2_system/server/build  ‹system_architecture*› 
╰─➤  cmake --build .                  
[ 14%] Building CXX object src/CMakeFiles/MeasuringServer.dir/Application/Application.cpp.o
In file included from /home/sagecat/Projects/measure_CO2_system/server/src/Application/Application.hpp:6,
                 from /home/sagecat/Projects/measure_CO2_system/server/src/Application/Application.cpp:1:
/home/sagecat/Projects/measure_CO2_system/server/src/Application/../Data/Data.hpp:9:42: error: ‘string’ is not a member of ‘std’
    9 | using Callback = std::function<void(std::string, Data)>;
      |                                          ^~~~~~
/home/sagecat/Projects/measure_CO2_system/server/src/Application/../Data/Data.hpp:1:1: note: ‘std::string’ is defined in header ‘<string>’; did you forget to ‘#include <string>’?
  +++ |+#include <string>
    1 | #ifndef DATA_HPP
/home/sagecat/Projects/measure_CO2_system/server/src/Application/../Data/Data.hpp:9:54: error: expression list treated as compound expression in functional cast [-fpermissive]
    9 | using Callback = std::function<void(std::string, Data)>;
      |                                                      ^
/home/sagecat/Projects/measure_CO2_system/server/src/Application/../Data/Data.hpp:9:55: error: template argument 1 is invalid
    9 | using Callback = std::function<void(std::string, Data)>;
      |                                                       ^
/home/sagecat/Projects/measure_CO2_system/server/src/Application/../Data/Data.hpp:9:23: error: ‘<expression error>’ in namespace ‘std’ does not name a type
    9 | using Callback = std::function<void(std::string, Data)>;
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/MeasuringServer.dir/build.make:76: src/CMakeFiles/MeasuringServer.dir/Application/Application.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:98: src/CMakeFiles/MeasuringServer.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
```

IhorBerehovskyi committed:
Fixed a build issues, added code formatting, renamed some fields

IhorBerehovskyi reviewed (commented):
For sure, good improvement

Sage-Cat committed:
Make working server. General refactoring

+ Refactor repo structure
+ Format repo using clang-format
+ Add SPDLOG logging

Signed-off-by: Sage-Cat <sagecatbox@gmail.com>

Sage-Cat commented on server/src/Database/SQLiteDatabase.cpp:
```diff
@@ -0,0 +1,14 @@
+SQLiteDatabase::SQLiteDatabase(const char *dbPath) : db(nullptr)
+{
+    if (sqlite3_open(dbPath, &db) != SQLITE_OK) {
+        // some code...
```
Всі коменти такого типу перейменуй у `// TODO: do this and this`

IhorBerehovskyi replied:
Done

Sage-Cat commented on server/src/CO2Sensor/CO2Sensor.hpp:
```diff
@@ -0,0 +1,46 @@
+    // Read CO2 concentration
+    int readCO2();
+
+    // ZERO POINT CALIBRATION
```
Це інфа гпт? Якщо так, то зроби її більше зрозумілою (що і як наш код має робити).
Пошукай в інеті можливо готові приклади

IhorBerehovskyi replied:
Done

Sage-Cat commented on server/src/CO2Sensor/CO2Sensor.hpp:
```diff
@@ -0,0 +1,46 @@
+#include <boost/asio.hpp>
+#include <string>
+
+class CO2Sensor {
```
Цю штуку я закоментив у мейні, бо треба спочатку протестити чи воно працює (підключити датчик і запустити на распбері)

IhorBerehovskyi replied:
Done

IhorBerehovskyi committed:
Refactored comments(now //TODO format is used, improved comments clarity in CO2Sensor.hpp), added variable in Data struct

IhorBerehovskyi reviewed (commented):
Looks good

IhorBerehovskyi committed:
Refactored variable name in Application
